apps/minifront/src/state/swap.ts

```diff
@@ -187,4 +200,48 @@ export const issueSwapClaim = async (swapCommitment: StateCommitment) => {
   await planBuildBroadcast('swapClaim', req, { skipAuth: true });
 };
 
+/*
+  Price impact is the change in price as a consequence of the trade's size. In SwapExecution, the \
+  first trace in the array is the best execution for the swap. To calculate price impact, take
+  the price of the trade and see the % diff off the best execution trace.
+ */
+const calculatePriceImpact = (swapExec?: SwapExecution): number | undefined => {
+  if (!swapExec?.traces.length || !swapExec.output || !swapExec.input) return undefined;
+
+  // Get the price of the estimate for the swap total
+  const inputAmount = getAmountFromValue(swapExec.input);
+  const outputAmount = getAmountFromValue(swapExec.output);
+  const swapEstimatePrice = divideAmounts(outputAmount, inputAmount);
+
+  // Get the price in the best execution trace
+  const inputAssetId = getAssetIdFromValue(swapExec.input);
+  const outputAssetId = getAssetIdFromValue(swapExec.output);
+  const bestTrace = swapExec.traces[0]!;
+  const bestInputAmount = getMatchingAmount(bestTrace.value, inputAssetId);
+  const bestOutputAmount = getMatchingAmount(bestTrace.value, outputAssetId);
+  const bestTraceEstimatedPrice = divideAmounts(bestOutputAmount, bestInputAmount);
+
+  // Difference = (priceB - priceA) / priceA
+  const percentDifference = swapEstimatePrice
+    .minus(bestTraceEstimatedPrice)
+    .div(bestTraceEstimatedPrice);
+
+  return percentDifference.toNumber();
+};
```

Comment:
@erwanor mind taking a look at this code section?

Reply:
This LGTM!